### PR TITLE
Add support for building on Windows and specifying custom boards

### PR DESCRIPTION
**Pull request by felipejfc**
(Draft PR for us to discuss the changes)

Adjusted paths so that the building will work on Windows/powershell. Also changed the default ncs version to 2.6.1 and fixed Zigbee with changes related to it. 
I've included BOARD_ROOT pointing to workdir on the generated CMakeLists so that one can use custom boards by creating a boards dir alongside zigbee.yaml file, I can create example related to it but it's a great way for us to achieve broader compatibility and adoption. I used this capability it to compile the binary to a nice!nano board and it worked great.

### CODE REVIEW

**Comment by felipejfc** [comment]
The explanation for filepath.Dir to path.Dir in some parts of template can be found in this issue https://github.com/golang/go/issues/44305#issuecomment-780111748

**Comment by ffenix113** [comment]
Hello @felipejfc,

Thank you for the PR! I looked through it and it does look good to me.
It is interesting that you didn't need to edit [this](https://github.com/ffenix113/zigbee_home/blob/86718037b1aabed130653c805a980f4cbd59e82c/runner/cmd.go#L101) function, as it provides some environment variables as well.

I will do some tests locally as well and post comments, if any, but generally have no objections for this PR.

`BOARD_ROOT` change is especially nice, as I also have a custom board to develop for, so this will help!

File: config/device.go
Comment (ffenix113) [inline config/device.go]:
```suggestion
				if runtime.GOOS == "windows" {
					return "C:\\ncs"
				}
				return "~/ncs"
```

When resolving NCS path in `GetToochainsPath` it will get value of this environment variable, and if it is not empty - will prefer it over that from configuration file.

The priority of the values will be (from highest to lowest):
- Value in env variable `NCS_TOOLCHAIN_BASE` (will be resolved in `GetToochainsPath`)
- Value defined in configuration file
- Default value, in this case `~/ncs` or `C:\ncs`, depending on OS.

File: config/device.go
Comment (ffenix113) [inline config/device.go]: This can be moved [here](https://github.com/ffenix113/zigbee_home/pull/72/files#r1676865077). If by this point there is no value set - this is something that user should address, as default configuration will always have the path.

File: templates/src/zigbee/carbon_dioxide.tpl
Comment (ffenix113) [inline templates/src/zigbee/carbon_dioxide.tpl:39]: This is a breaking change, but it should be okay. Will mention this in release notes.

**Comment by felipejfc** [comment]
> It is interesting that you didn't need to edit [this](https://github.com/ffenix113/zigbee_home/blob/86718037b1aabed130653c805a980f4cbd59e82c/runner/cmd.go#L101) function, as it provides some environment variables as well.

Thank you very much for reviewing!
Regarding this:

> It is interesting that you didn't need to edit [this](https://github.com/ffenix113/zigbee_home/blob/86718037b1aabed130653c805a980f4cbd59e82c/runner/cmd.go#L101) function, as it provides some environment variables as well.

The reason I didn't have problems is probably that whenever I'm running zigbee_home binary, I'm doing it from inside a nrf shell environment with: ```nrfutil toolchain-manager launch --shell ``` 
It would probably be better if we also fix this method for windows so that running from inside the shell is not a requirement. But I'm not sure what the best approach would be, maybe make zb home source all env vars on the file $NCS_TOOLCHAIN_BASE/toolchains/$SELECTED_TOOLCHAIN/environment.json? What do you think?

**Comment by felipejfc** [comment]
Update: pushed a new commit fixing the file you pointed out and some more path separators. Now I can compile a firmware successfully even directly in powershell without needing to enter an ncs terminal first

File: templates/src/zigbee/carbon_dioxide.tpl
Comment (ffenix113) [reply templates/src/zigbee/carbon_dioxide.tpl:39]: On the second thought this change is not major enough, and in configuration we have some functionality to [parse & compare versions](https://github.com/ffenix113/zigbee_home/blob/86718037b1aabed130653c805a980f4cbd59e82c/config/ncs_finder.go#L112), which could be used as a guard here as well. After this is merged, I will add the checks to make it compatible with <v2.6.0 & >= v2.6.0. :+1:

File: runner/cmd.go
Comment (ffenix113) [inline runner/cmd.go:103]: Oh, nice, didn't know about this. Definitely worth building env on top of this!

**Comment by ffenix113** [comment]
Could you please remove [this](https://github.com/ffenix113/zigbee_home/pull/72#discussion_r1676865883), and I am ready to approve & merge this, unless you have anything else to discuss.

Great work!

**Comment by felipejfc** [comment]
> Could you please remove [this](https://github.com/ffenix113/zigbee_home/pull/72#discussion_r1676865883), and I am ready to approve & merge this, unless you have anything else to discuss.
> 
> Great work!

done!

**Review by ffenix113 — APPROVED** [review]
Great work, thank you!

**Pull request merged**
